doc/contributing/maintaining-shared-library-support.md

```diff
@@ -0,0 +1,121 @@
+# Maintaining shared library support
+
+Node.js unofficially supports a build option where Node.js is built as
+a shared library. The shared library is node.dll on windows, and
+libnode on other platforms.
+The shared library provides a way to embed Node.js into other
+applications and to have multiple applications use a single copy of
+Node.js instead of having to bundle in the full Node.js footprint
+into each application. For workloads that require multiple Node.js
+instances this can result in a significant footprint savings.
+
+This document provides an outline of the approach and things to look
+out for when maintaining the shared library support.
+
+Currently, shared library support has only been tested on:
+
+* Linux
+* macOS
+* Windows
+* AIX
+
+## Building with shared library option
+
+On non-Windows platoforms, Node.js is built with the shared library
+option by adding `--shared` to the configure step. On Windows
+platofrms Node.js is built with the shared library option by
+adding `dll` to the vcbuild command line.
+
+Once built there are two key components:
+
+* executable - node
+* library - libnode
+
+The node executable is a thin wrapper around libnode which is
+generated so that we can run the standard Node.js test suite
+against the shared library.
+
+The executable and library will have extensions as appropriate
+for the platform on which they are built. For
+example, node.exe on windows and node on other platforms for
+the executable.
+
+libnode may have additional naming components, as an example
+in a build on macOS `libnode.105.dylib`.
+
+In cases where an application links against the shared
+library it is up to the application developer to add options
+so that the shared library can be found by the application or
+to set the LIBPATH (AIX), LD\_LIBRARY\_PATH (Linux/Unix), etc.
+so that it is found at runtime.
+
+For the node.exe wrapper, it is built so that it can
+find the shared library in one of the following:
+
+* the same directory as the node executable
+* ../lib with the expectation that the executable is
+  installed in a `bin` directory at the same level
+  as a `lib` directory in which the shared library is
+  installed. This is where the default package that
+  is build with the shared library option will
+  place the executable and library.
```

Comment:
I think this is only true on Linux and macOS where we can embed a relative RPATH into the executable. AIX doesn't have this luxury so currently it hard codes an _absolute_ path to libnode.a into the executable.

On Windows libnode.dll and node.exe sit in the same directory as this is the expected location for DLLs. See [this MSDN article](https://docs.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-search-order#search-order-for-desktop-applications) if you want to know more about the DLL search order on Windows.

Reply:
@lux01 updated, can you tell me what the hardcoded path for AIX is so that I can add it?